### PR TITLE
maxranges, fix #88

### LINKED ISSUE #88: [folding] limit number of ranges created
The LSP folding API provides a number of max ranges the that the server should create.
We should pass that limit to the service and limit the number of ranges

### PR DESCRIPTION
**Pull request by octref**

### CODE REVIEW

File: src/services/cssFolding.ts [hunk 1]
```diff
@@ -79,11 +79,24 @@ export function getFoldingRegions(document: TextDocument): FoldingRangeList {
 					}
 
 					if (startLine !== endLine) {
-						ranges.push({
-							startLine,
-							endLine,
-							type: undefined
-						});
+						if (maxRanges) {
```
Comment (aeschli) [inline src/services/cssFolding.ts]: Of course we can just cut the last regions, but that is not the best user experience.
In HTML and JSON implementatio I filter out the most nested ranges.

File: src/cssLanguageService.ts [hunk 2]
```diff
@@ -94,7 +94,7 @@ export interface LanguageService {
 	findDocumentColors(document: TextDocument, stylesheet: Stylesheet): ColorInformation[];
 	getColorPresentations(document: TextDocument, stylesheet: Stylesheet, color: Color, range: Range): ColorPresentation[];
 	doRename(document: TextDocument, position: Position, newName: string, stylesheet: Stylesheet): WorkspaceEdit;
-	findFoldingRegions(document: TextDocument, stylesheet: Stylesheet): FoldingRangeList;
+	findFoldingRegions(document: TextDocument, maxRanges?: number | undefined): FoldingRangeList;
```
Comment (aeschli) [inline src/cssLanguageService.ts]: Note that this is a breaking API change and will require an increase of a major version update.

Comment (aeschli) [reply src/cssLanguageService.ts]: Oh, we are on a next release. No problem then!

**Comment by aeschli** [comment]
merged, we can implement a better filter strategy later

File: src/services/cssFolding.ts (hunk 1 above)
Comment (octref) [reply src/services/cssFolding.ts]: Extracted to #90.

**Pull request merged**